oidb/pepper/common/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright © Aptos Foundation
+
+use crate::asymmetric_encryption::AsymmetricEncryption;
+use aes_gcm::aead::rand_core::{CryptoRng as AeadCryptoRng, RngCore as AeadRngCore};
+use anyhow::bail;
+use curve25519_dalek::digest::Digest;
+use rand_core::{CryptoRng, RngCore};
+use serde::{Deserialize, Serialize};
+
+pub mod asymmetric_encryption;
+pub mod elgamal;
+pub mod jwt;
+pub mod vrf;
+
+pub fn sha3_256(input: &[u8]) -> Vec<u8> {
+    let mut hasher = sha3::Sha3_256::new();
+    hasher.update(input);
+    hasher.finalize().to_vec()
+}
+
+/// The spec of a request to this pepper service.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct PepperRequest {
+    pub jwt_b64: String,
+    /// If specified, generate pepper for `jwk.payload.iss, jwk.payload.sub, overriding_aud`.
+    /// Otherwise, generate pepper for `jwk.payload.iss, jwk.payload.sub, jwk.payload.aud`.
+    pub overriding_aud: Option<String>,
+    pub epk_hex_string: String,
+    pub epk_expiry_time_secs: u64,
+    pub epk_blinder_hex_string: String,
+    pub uid_key: Option<String>,
+}
+
+/// The response to `PepperRequestV0`, which contains the calculated pepper (hexlified) or a processing error.
+#[derive(Debug, Deserialize, Serialize)]
+pub struct PepperResponse {
+    pub pepper_key_hex_string: String,
+    pub pepper_hex_string: String,
+}
+
+/// The response to `PepperRequestV1`, which contains the calculated pepper (encrypted then hexlified) or a processing error.
+#[derive(Debug, Deserialize, Serialize)]
+pub enum PepperResponseV1 {
+    OK { pepper_encrypted_hexlified: String },
+    Error(String),
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct VRFVerificationKey {
+    pub scheme_name: String,
+    pub vrf_public_key_hex_string: String,
+}
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct EncryptionPubKey {
```

Comment:
Is this even used anywhere?

Reply:
no, since we are not doing encryption at the moment.

Reply:
Will it be used?

Reply:
now i remember this can be gone, given `EphemeralPublicKey` in main.

Reply:
fixed